python3Packages.pyjwt: revert approach to conditionally disable test in override; home-assistant: fix build

mweinelt opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

Fixing up #122380 … BUT it doesn't seem to work. The test does not get enabled when version is overwritten.

```
=========================== short test summary info ============================
FAILED tests/contrib/test_algorithms.py::TestEcdsaAlgorithms::test_ec_verify_should_return_false_if_signature_invalid
====== 1 failed, 178 passed, 8 skipped, 1 xfailed, 140 warnings in 2.01s =======
builder for '/nix/store/q8jhyxiksz54n85sbnrbr05n6ggbxqd1-python3.8-pyjwt-1.7.1.drv' failed with exit code 1
```

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

mweinelt commented:
@GrahamcOfBorg build home-assistant

mweinelt commented:
I was told the check was already evaluated at the time the override happend, so I say we split off pyjwt 1.7.1 as pyjtw_1 and migrate packages that need it to that.

mweinelt committed:
Revert "python3Packages.pyjwt: disable failing test for 1.7.1"

This reverts commit b56bd9afccf63152eb59f891863458049aa4a8e3.

andir commented on pkgs/development/python-modules/pyjwt/default.nix:
```diff
@@ -31,7 +31,7 @@ buildPythonPackage rec {
 
   disabledTests = lib.optionals (lib.versionOlder version "2.0") [
```
How is this supposed to work? The expression will always be eval'ed against the hardcoded `version` further up the the file. Overrides will not replace this and also not magically change the way `version` is bound (e.g. via `rec`).

Example on why this doesn't work:

Lets say you have a function: 
```
a: rec {
  version = "some-version";
  something = if version == "123" then "foo" else version;
}
```

The result will always be `{ version = "some-version"; something = foo"; }`.

If you could `overrideAttrs (_: { version = "123"; })` on the above it would desugar to something like this:

```
rec {
  version = "some-version";
  something = if version == "123" then "foo" else version;
} // {
  version = "123";
}
```

Which equals `{ version = "123"; something = "some-version"; }`.

mweinelt replied:
Thanks for the explanation.

mweinelt committed:
home-assistant: fix build by disabling test in pyjwt override

FAILED tests/contrib/test_algorithms.py::TestEcdsaAlgorithms::test_ec_verify_should_return_false_if_signature_invalid

fabaff reviewed (approved):
Thanks cleaning-up my mess.
